lib/sort_alphabetical.rb

```diff
@@ -19,8 +19,12 @@ def sort(set)
   end
 
   def normalize(string)
-    UnicodeUtils.compatibility_decomposition(string).split(''.freeze).select do |c|
-      UnicodeUtils.general_category(c) =~ /Letter|Separator|Punctuation|Number/
-    end.join
+    result = String.new
+    UnicodeUtils.compatibility_decomposition(string).each_char do |c|
+      if UnicodeUtils.general_category(c) =~ /Letter|Separator|Punctuation|Number/
```

Comment:
a good improvement might be making this a `good.include?(UnicodeUtils.general_category(c))`